atomicfu/src/concurrentTest/kotlin/kotlinx/atomicfu/locks/ThreadParkerTest.kt

```diff
@@ -0,0 +1,57 @@
+package kotlinx.atomicfu.locks
+
+import kotlinx.atomicfu.atomic
+import kotlin.test.Test
+import kotlin.test.assertTrue
+import kotlin.time.Duration
+
+class ThreadParkerTest {
+
+
+    @Test
+    fun parkUnpark() {
+        var parkingHandle: ParkingHandle? = null
+        var done = false
+        
+        testThread { 
+            parkingHandle = ParkingSupport.currentThreadHandle()
+            ParkingSupport.park(Duration.INFINITE)
+            done = true
+        }
+        
+        // Allow thread to be parked before park call
+        sleepMillis(100)
+        ParkingSupport.unpark(parkingHandle!!)
+        
+        // Allow thread to wakeup
+        sleepMillis(400)
+        assertTrue(done)
```

Comment:
Why not use `Fut` here instead of `testThread` and finish the test exactly when the thread stops? I suspect it takes much less than half a second to unpark a thread.